woocommerce/woocommerce
# Order product count api requests orders store

### All Submissions:






### Changes proposed in this Pull Request:

Adds a new orders data store that makes use of the orders API -> https://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-orders
For now this only allows for retrieving/search for orders and getting the orders count.

Partially addresses #32524 

### How to test the changes in this Pull Request:

1. Load this branch and create a couple of orders and go to **WooCommerce > Home**
2. Open your console and run `wp.data.select('wc/admin/orders').getOrders({ status: 'completed'}, undefined);`
3. It should return `undefined` at first
4. Run it again, assuming the request finished, and it should return all the completed orders. You can try the above again with other items in the query, like `_fields: ['id']` for example, which should only return the `ids`, any query param defined here should work: https://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-orders
5. Now do the same for the `getOrdersTotalCount` function, by running: `wp.data.select('wc/admin/orders').getOrdersTotalCount({ status: 'completed'}, undefined);`, this might run the correct number right away if you recently ran `getOrders` with the same query params, otherwise it should return `undefined` initially.
6. If you have the [Redux Devtools](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en) extension, you can look for the `wc/admin/orders` store and make sure the state is updated correctly and the actions are triggered correctly.

### Other information:




### FOR PR REVIEWER ONLY:
